libs/vm_monitor/src/dispatcher.rs

```diff
@@ -83,18 +83,24 @@ impl Dispatcher {
         let highest_shared_version = match monitor_range.highest_shared_version(&agent_range) {
             Ok(version) => {
                 sink.send(Message::Text(
-                    serde_json::to_string(&ProtocolResponse::Version(version)).unwrap(),
+                    Utf8Bytes::try_from(
+                        serde_json::to_vec(&ProtocolResponse::Version(version)).unwrap(),
+                    )
+                    .unwrap(),
                 ))
                 .await
                 .context("failed to notify agent of negotiated protocol version")?;
                 version
             }
             Err(e) => {
                 sink.send(Message::Text(
-                    serde_json::to_string(&ProtocolResponse::Error(format!(
-                        "Received protocol version range {} which does not overlap with {}",
-                        agent_range, monitor_range
-                    )))
+                    Utf8Bytes::try_from(
```

Comment:
Utf8Bytes [impls try-less from](https://docs.rs/axum/latest/axum/extract/ws/struct.Utf8Bytes.html#impl-From%3CString%3E-for-Utf8Bytes) a String, I think that's better than going through a byte array.